server/testutils/factories/application.ts

```diff
@@ -33,6 +33,16 @@ class ApplicationFactory extends Factory<ApprovedPremisesApplication> {
       },
     })
   }
+
+  withPageResponse(section: string, task: string, key: string, value: unknown) {
```

Comment:
❤️ Love it!

Reply:
Wonder if named parameters would make this clearer. Something like:

```
withPageResponse(args: {section: string, task: string, key: string, value: unknown})
```

Reply:
Yeah good point!